### PR TITLE
Initial implementation of github-nginx-cache

### PR DESCRIPTION
**Pull request by billytrend**

### CODE REVIEW

File: kubernetes-config/loadbalancer.yml [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+metadata:
+  name: github-nginx-cache-loadbalancer
+spec:
+  ports:
+    - port: 80
+      protocol: TCP
+      targetPort: 80
+  selector:
+    app: github-nginx-cache
+  type: LoadBalancer
```
Comment (vladbarosan) [inline kubernetes-config/loadbalancer.yml]: these are just samples i presume, the actual specs will be in the deployment repo   ( also why do we need type loadBalancer ) ?

Comment (billytrend) [reply kubernetes-config/loadbalancer.yml]: yeah 👍

Comment (billytrend) [reply kubernetes-config/loadbalancer.yml]: Loadbalancer was just for testing

**Review by vladbarosan — APPROVED** [review]

**Comment by vladbarosan** [comment]
whi is the task running on the PR ?

**Comment by vladbarosan** [comment]
also make sure to add one that publishes to the internal ACR

**Comment by billytrend** [comment]
i think it just automatically picked up the tasks when i ran them manually, one of them does publish to the private acr too

File: nginx-config/health_check.conf [hunk 2]
```diff
@@ -0,0 +1,4 @@
+location /health/alive {
+    return 200 '{ "message": "Github cache is alive" }';
+    add_header Content-Type text/plain;
```
Comment (vladbarosan) [inline nginx-config/health_check.conf]: application json ?

**Review by vladbarosan — APPROVED** [review]

File: kubernetes/active-passive/check-readiness.sh [hunk 3]
```diff
@@ -15,11 +15,11 @@ podhealth() {
   curl -I $activepod
 
   # if curl fails then we need to be active otherwise stay passive
-  if [[ $? -ne 0 ]]
+  if [[ "$?" -eq "0" ]]
```
Comment (vladbarosan) [inline kubernetes/active-passive/check-readiness.sh]: why do you need to make strings out of these ?

Comment (billytrend) [reply kubernetes/active-passive/check-readiness.sh]: because b a s h

File: kubernetes/active-passive/check-readiness.sh [hunk 4]
```diff
@@ -10,9 +10,11 @@ podhealth() {
 
   activepod=$(for i in 0 1; do echo $podname-$i;done | grep -v $HOSTNAME)
 
-  echo "activepod = ${activepod}"
+  echo "activepod = ${activepod}.${serviceDomain}"
```
Comment (vladbarosan) [inline kubernetes/active-passive/check-readiness.sh:13]: where is service domain coming from ? is it K8 populated ?

Comment (billytrend) [reply kubernetes/active-passive/check-readiness.sh:13]: no this is a custom env variable.. do you know if there are defaults that k8s provides? had a look around and couldn't find a better way to get this

File: README.md [hunk 5]
```diff
@@ -1,10 +1,25 @@
-### Build
+## Github nginx cache
+
+This repo contains nginx configuration tuned to sit in front of github endpoints and provide caching functionality. Github will not rate-limit [conditional requests](localhost:8000/api/repos/azure/github-nginx-cache). The [`proxy_cache_*` nginx directives](http://nginx.org/en/docs/http/ngx_http_proxy_module.html#proxy_cache) force nginx to revalidate any cached content from the upstream server (in this case, github). Revalidation is performed by github as a conditional request, therefore it will not reduce api limits. This works for both authenticated and unauthenticated requests.
```
Comment (vladbarosan) [inline README.md]: this should probably point to the github docs ?

File: README.md [hunk 6]
```diff
@@ -18,6 +33,13 @@
     docker build -t custom-nginx . && docker run -it -p 8000:80  -v $(pwd)/nginx-logs:/var/log/nginx custom-nginx
     curl localhost:8000/health/alive
 
+### Build
```
Comment (vladbarosan) [inline README.md]: 2 build sections

File: kubernetes/example-config/pod.yml [hunk 7]
```diff
@@ -6,7 +6,7 @@ metadata:
     app: github-nginx-cache
 spec:
   containers:
-    - image: azuredevx/github-nginx-cache:test
+    - image: azuredevx/github-nginx-cache:latest
```
Comment (vladbarosan) [inline kubernetes/example-config/pod.yml]: maybe add a warning to use latest only for dev purposes. for prod use a specific release tag

**Review by vladbarosan — APPROVED** [review]
Approved with suggestions

**Pull request merged**